src/components/cart/cartList.js

```diff
@@ -0,0 +1,224 @@
+import { useCart } from "./cartContext";
+import { useWishList } from "../WishList/wishContext";
+import { Header } from "../header";
+import { BACKEND_URL } from "../backendUrl";
+import axios from "axios";
+import "./cartStyles.css";
+import { useState } from "react";
+import { ToastContainer, toast } from "react-toastify";
+import { DataLoader } from "../DataLoader";
+
+export const Cart = () => {
+  const { itemsInCart, dispatch: cartDispatch } = useCart();
+  const { dispatch: wishDispatch, wishList } = useWishList();
+  const [isQtyUpdated, setQtyUpdate] = useState(true);
+  const totalReducer = () =>
+    itemsInCart.reduce((acc, item) => acc + item.price * item.quantity, 0);
+
+  const CartUpdate = async ({ type, payLoad }) => {
+    const _id = payLoad;
+    const itemFound = itemsInCart.find((item) => item._id === _id);
+    let updatedQuantity;
+    if (type === "INCREMENT") {
+      updatedQuantity = itemFound.quantity + 1;
+    } else if (type === "DECREMENT") {
+      updatedQuantity = itemFound.quantity - 1;
+      // return updatedQuantity === 0 && deleteCartItem(_id);
+      if (updatedQuantity === 0) {
+        return deleteCartItem(_id);
+      }
+    }
+    try {
+      setQtyUpdate(false);
+      const { data } = await axios.post(`${BACKEND_URL}cart/${_id}`, {
+        _id: _id,
+        quantity: updatedQuantity
+      });
+      payLoad = {
+        _id: _id,
+        quantity: updatedQuantity
+      };

```

Comment:
You can omit this and use the updated response coming from the backend as a payload